lang/test/org/partiql/lang/eval/evaluatortestframework/PipelineEvaluatorTestAdapter.kt

```diff
@@ -1,136 +1,31 @@
-fun EvaluatorErrorTestCase.testDetails(
-    note: String,
-    actualErrorCode: ErrorCode? = null,
-    actualErrorContext: PropertyValueMap? = null,
-    actualPermissiveModeResult: String? = null,
-    actualInternalFlag: Boolean? = null,
-): String {
-    val b = StringBuilder()
-    b.appendLine("Note                           : $note")
-    b.appendLine("Group name                     : $groupName")
-    b.appendLine("Query                          : $query")
-    b.appendLine("Expected error code            : $expectedErrorCode")
-    if (actualErrorCode != null) {
-        b.appendLine("Actual error code              : $actualErrorCode")
-    }
-    b.appendLine("Expected error context         : $expectedErrorContext")
-    if (actualErrorContext != null) {
-        b.appendLine("Actual error context           : $actualErrorContext")
-    }
-    b.appendLine("Expected internal flag         : $expectedInternalFlag")
-    if (actualErrorContext != null) {
-        b.appendLine("Actual internal flag           : $actualInternalFlag")
-    }
-    b.appendLine("Expected permissive mode result: $expectedPermissiveModeResult")
-    if (actualPermissiveModeResult != null) {
-        b.appendLine("Actual permissive mode result  : $actualPermissiveModeResult")
-    }
-    return b.toString()
-}
-
-private fun assertEquals(
-    expected: Any?,
-    actual: Any?,
-    reason: EvaluatorTestFailureReason,
-    detailsBlock: () -> String
-) {
-    if (expected != actual) {
-        throw EvaluatorAssertionFailedError(reason, detailsBlock())
-    }
-}
-
-private fun <T> assertDoesNotThrow(
-    reason: EvaluatorTestFailureReason,
-    detailsBlock: () -> String,
-    block: () -> T
-): T {
-    try {
-        return block()
-    } catch (ex: Throwable) {
-        throw EvaluatorAssertionFailedError(reason, detailsBlock(), ex.cause)
-    }
-}
-
-private inline fun assertThrowsSqlException(
-    reason: EvaluatorTestFailureReason,
-    detailsBlock: () -> String,
-    block: () -> Unit
-): SqlException {
-    try {
-        block()
-        // if we made it here, the test failed.
-        throw EvaluatorAssertionFailedError(reason, detailsBlock())
-    } catch (ex: SqlException) {
-        return ex
-    }
-}
-
-class AstEvaluatorTestAdapter : EvaluatorTestAdapter {
+internal class PipelineEvaluatorTestAdapter(
+    private val pipelineFactory: PipelineFactory
+) : EvaluatorTestAdapter {
 
     override fun runEvaluatorTestCase(tc: EvaluatorTestCase, session: EvaluationSession) {
-        if (tc.implicitPermissiveModeTest) {
-            val testOpts = CompileOptions.build { tc.compileOptionsBuilderBlock(this) }
-            assertNotEquals(
-                TypingMode.PERMISSIVE, testOpts.typingMode,
-                "Setting TypingMode.PERMISSIVE when implicit permissive mode testing is enabled is redundant"
-            )
+        if (tc.target != EvaluatorTestTarget.ALL_PIPELINES && pipelineFactory.target != tc.target) {
+            return
```

Comment:
Why do we return here if `tc.target != EvaluatorTestTarget.ALL_PIPELINES` and `pipelineFactory.target != tc.target`?

Reply:
This is what skips execution of the test.

Reply:
Ah ok, that explains things.